### PR TITLE
fix(layers): Fix incorrect masked mean/variance in BatchNormalization layer

### LINKED ISSUE #19848: BatchNormalization gives incorrect output with masked inputs > 3 dimensions
The mean/variance calculations are incorrect, which means the inputs are not normalized correctly. E.g.

``` python
import keras

x = keras.ops.ones((1, 2, 3, 4))
x._keras_mask = keras.ops.ones((1, 2, 1))

y = keras.layers.BatchNormalization()(x, training=True)

print(keras.ops.mean(y, axis=-1))
```
gives output
```
tf.Tensor([-0.57732624 -0.57732624 -0.57732624 -0.57732624], shape=(4,), dtype=float32)
```
instead of the correct normalized output (`[0, 0, 0, 0]`).

The basic issue is that this calculation is incorrect:
https://github.com/keras-team/keras/blob/efaaf85e19113400f23462cbafcef433cd95ad9c/keras/src/layers/normalization/batch_normalization.py#L310-L314
because it doesn't account for the broadcasting (i.e. it gives a value of 2 in the above example, when it should be 2 * 3 * 4).

See https://github.com/keras-team/keras/issues/19818 for more discussion/background.

### PR DESCRIPTION
**Pull request by harshaljanjani**
- Update masked moments calculation to properly account for broadcast dimensions when summing mask weights.
- Added test to verify broadcast mask handling produces zero-centered outputs.
- Fixes #19848.

### CODE REVIEW

**Comment by harshaljanjani** [comment]
The error is in the line `ops.mean()` for the OpenVINO backend:
`axis_const = ov_opset.constant(axis, dtype=Type.i32).output(0)`
Until further information is available, I will mark the test case as skipped for the OpenVINO backend.

File: keras/src/layers/normalization/batch_normalization.py [hunk 1]
```diff
@@ -314,39 +314,44 @@ def _moments(self, inputs, mask):
         if mask is None:
             return ops.moments(
                 inputs,
-                axes=self._reduction_axes,
+                axes=self._reduction_axes
+                if isinstance(self._reduction_axes, list)
```
Comment (fchollet) [inline keras/src/layers/normalization/batch_normalization.py:318]: Can we normalize `self._reduction_axes` to a list earlier (when we create the attr)?

**Review by fchollet — COMMENTED** [review]
Thanks for the fix!

File: keras/src/layers/normalization/batch_normalization.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def build(self, input_shape):
 
         reduction_axes = list(range(len(input_shape)))
         del reduction_axes[self.axis]
-        self._reduction_axes = reduction_axes
+        self._reduction_axes = list(reduction_axes)
```
Comment (fchollet) [inline keras/src/layers/normalization/batch_normalization.py:217]: As per above it's already always a list?

Comment (harshaljanjani) [reply keras/src/layers/normalization/batch_normalization.py]: Fixed, apologies for the oversight. It turns out all the validation checks weren’t needed in the first place.

**Comment by harshaljanjani** [comment]
@fchollet I know it's been a long PR with the `RandomChoice` and `RandomApply` layers, but it's my first month contributing, and I've learned a lot about `src/layers/preprocessing` during the migration process. It's alright if you don't want those layers in the codebase, but I'll fix the reviews and re-request a review anyway, thanks!

**Review by fchollet — APPROVED** [review]
LGTM, thank you
